src/executors/public_1559_executor.rs

```diff
@@ -73,27 +73,43 @@ where
 
         action.execution.tx.set_from(address);
 
-        // redundant match to log specific reasons
+        // early return on OrderAlready filled
         // always use 1_000_000 gas for now
-        let gas_usage_result = self
+        let gas_usage_result: Result<U256, anyhow::Error> = self
             .client
             .estimate_gas(&action.execution.tx, None)
             .await
-            .unwrap_or_else(|err| {
+            .or_else(|err| {
                 if let Some(Value::String(four_byte)) =
                     err.as_error_response().unwrap().data.clone()
                 {
-                    warn!(
-                        "Error estimating gas with reason: {}; {}",
-                        Into::<ReactorErrorCode>::into(four_byte.clone()),
-                        four_byte
-                    );
+                    let error_code = ReactorErrorCode::from(four_byte.clone());
+                    match error_code {
+                        ReactorErrorCode::OrderAlreadyFilled => {
+                            info!("{} - Order already filled, skipping execution", order_hash);
+                            Err(anyhow::anyhow!("Order Already Filled"))
+                        }
```

Comment:
We should return early on InvalidDeadline too

Reply:
true true